### PR TITLE
Added Gaudi2 CI tests.

### PR DESCRIPTION
**Pull request by vshekhawat-hlab**

### CODE REVIEW

**Comment by vshekhawat-hlab** [comment]
HI @loadams ,

Updated hpu-gaudi2 to run only the required tests defined in the test_list. However, I noticed that verification checks are not being triggered, as the maintainer needs to approve them. Could you please approve it so that I can check if the test_list-based method is working in the YAML file? Or do you have any other method to execute specific tests.

File: .github/workflows/hpu-gaudi2.yml [hunk 1]
```diff
@@ -38,6 +91,13 @@ jobs:
...
+      - name: Install transformers
+        run: |
+          git clone https://github.com/huggingface/transformers
+          cd transformers
+          git rev-parse --short HEAD
+          pip install .
+
       - name: Install deepspeed
         run: |
           pip install .[dev]
```
Comment (loadams) [inline .github/workflows/hpu-gaudi2.yml:103]: Should just need to add autotuning here to get the remaining tests to pass

Comment (vshekhawat-hlab) [reply .github/workflows/hpu-gaudi2.yml]: Thanks @loadams ,

Updated PR.

File: .github/workflows/hpu-gaudi2.yml [hunk 2]
```diff
@@ -26,6 +26,58 @@ jobs:
         - 80
       options: --runtime=habana -e HABANA_VISIBLE_DEVICES=all -e OMPI_MCA_btl_vader_single_copy_mechanism=none --cap-add=sys_nice
 
+    env:
+      PT_HPU_LAZY_MODE: 0
+      TEST_LIST: |
```
Comment (loadams) [inline .github/workflows/hpu-gaudi2.yml:31]: Tagging @mrwyattii for his thoughts on running the test list this way vs trying to skip tests that are unsupported.

Comment (vshekhawat-hlab) [reply .github/workflows/hpu-gaudi2.yml:31]: @loadams @mrwyattii ,

Any update on the PR. Should we merge this?

Comment (loadams) [reply .github/workflows/hpu-gaudi2.yml:31]: Hi @vshekhawat-hlab - we are working on fixing the other CI tests then we will get this merged.

Comment (loadams) [reply .github/workflows/hpu-gaudi2.yml:31]: These should be resolved now, thanks!

**Review by mrwyattii — APPROVED** [review]

**Pull request merged**